--- xv6-clean/sysproc.c	2026-08-24 14:26:16.859049500 +0000
+++ asg7/q4-sched/sysproc.c	2026-10-06 13:54:18.029437500 +0000
@@ -89,3 +89,21 @@
   release(&tickslock);
   return xticks;
 }
+
+int
+sys_set_priority(void)
+{
+  int pid, prio;
+  if(argint(0, &pid) < 0 || argint(1, &prio) < 0)
+    return -1;
+  return set_priority(pid, prio);
+}
+
+int
+sys_get_priority(void)
+{
+  int pid;
+  if(argint(0, &pid) < 0)
+    return -1;
+  return get_priority(pid);
+}
